t/plugin/fault-injection.t

```diff
@@ -0,0 +1,562 @@
+                               "proxy-rewrite": {
+                                   "uri": "/hello"
+                               }
+                           },
+                           "upstream": {
+                               "nodes": {
+                                   "127.0.0.1:1980": 1
+                               },
+                               "type": "roundrobin"
+                           },
+                           "uri": "/hello"
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 400
+--- response_body eval
+qr/validation failed: failed to match pattern/
+--- no_error_log
+[error]
+
+
+
+=== TEST 7: set route(invalid duration with whitespace in the delay property)
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                           "plugins": {
+                               "fault-injection": {
+                                   "delay": {
+                                       "duration": "0. 1"
+                                   }
+                               },
+                               "proxy-rewrite": {
+                                   "uri": "/hello"
+                               }
+                           },
+                           "upstream": {
+                               "nodes": {
+                                   "127.0.0.1:1980": 1
+                               },
+                               "type": "roundrobin"
+                           },
+                           "uri": "/hello"
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 400
+--- response_body eval
+qr/validation failed: failed to match pattern/
+--- no_error_log
+[error]
+
+
+
+=== TEST 8: set route(delay 1 seconds)
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                           "plugins": {
+                               "fault-injection": {
+                                   "delay": {
+                                       "duration": "1"
+                                   }
+                               },
+                               "proxy-rewrite": {
+                                   "uri": "/hello"
+                               }
+                           },
+                           "upstream": {
```

Comment:
we can remove `upstream` object, it is useless. please fix some other similar points.